# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/ffmpeg/intel-patches/0021-lavc-decode-Add-get_hw_config-function.patch
# Copyright (C) 2023 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

From 92be7fa47898d46ac23aab21d6c0f04519c4b482 Mon Sep 17 00:00:00 2001
From: Linjie Fu <linjie.fu@intel.com>
Date: Thu, 2 Jan 2020 11:06:45 +0800
Subject: [PATCH 21/77] lavc/decode: Add get_hw_config function

Wrap the procedure of getting the hardware config from a pixel format
into a function.

Signed-off-by: Linjie Fu <linjie.fu@intel.com>
---
 libavcodec/decode.c | 33 +++++++++++++++++++++------------
 1 file changed, 21 insertions(+), 12 deletions(-)

diff --git a/libavcodec/decode.c b/libavcodec/decode.c
index 1893caa6a6..bfa82199ae 100644
--- a/libavcodec/decode.c
+++ b/libavcodec/decode.c
@@ -1097,6 +1097,26 @@ static void hwaccel_uninit(AVCodecContext *avctx)
     av_buffer_unref(&avctx->hw_frames_ctx);
 }
 
+static const AVCodecHWConfigInternal *get_hw_config(AVCodecContext *avctx, enum AVPixelFormat fmt)
+{
+    const AVCodecHWConfigInternal *hw_config;
+    int i;
+
+    if (ffcodec(avctx->codec)->hw_configs) {
+        for (i = 0;; i++) {
+            hw_config = ffcodec(avctx->codec)->hw_configs[i];
+            if (!hw_config)
+                break;
+            if (hw_config->public.pix_fmt == fmt)
+                break;
+        }
+    } else {
+        hw_config = NULL;
+    }
+
+    return hw_config;
+}
+
 int ff_get_format(AVCodecContext *avctx, const enum AVPixelFormat *fmt)
 {
     const AVPixFmtDescriptor *desc;
@@ -1154,18 +1174,7 @@ int ff_get_format(AVCodecContext *avctx, const enum AVPixelFormat *fmt)
             break;
         }
 
-        if (ffcodec(avctx->codec)->hw_configs) {
-            for (i = 0;; i++) {
-                hw_config = ffcodec(avctx->codec)->hw_configs[i];
-                if (!hw_config)
-                    break;
-                if (hw_config->public.pix_fmt == user_choice)
-                    break;
-            }
-        } else {
-            hw_config = NULL;
-        }
-
+        hw_config = get_hw_config(avctx, user_choice);
         if (!hw_config) {
             // No config available, so no extra setup required.
             ret = user_choice;
-- 
2.17.1

